app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/BaseViewModel.kt

```diff
@@ -22,20 +24,43 @@ import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import timber.log.Timber
 
 /** Provides [LiveEvent]s for both navigation and external navigation and helper functionality to observe the [LiveData] from a [Fragment] */
 abstract class BaseViewModel : ViewModel(), KoinComponent {
     protected val dispatcherProvider: DispatcherProvider by inject()
+    private val toaster: Toaster by inject()
 
     /**
      * Helper to launch to IO thread quickly
      */
     fun launchIO(block: suspend CoroutineScope.() -> Unit): Job =
         viewModelScope.launch(dispatcherProvider.IO, block = block)
 
+    suspend fun runOnMain(block: suspend CoroutineScope.() -> Unit) =
+        withContext(dispatcherProvider.Main, block)
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Error handling
+    ///////////////////////////////////////////////////////////////////////////
+    suspend fun handleError(messageId: Int) {
```

Comment:
@BottleRocket-Colin On the fence about this one. We should either:

1. Annotate all messageId: Int function arguments with the `@StringRes` qualifier to help lint catch incorrect usages.
2. Change the messageId type to be a `StringIdHelper`, which is more flexible by accepting raw strings, format strings, or plurals in addition to String id resources.

If we only ever use string id resources, approach 1 is fine. If we wanted to support additional types of error messages now, we could use approach 2.

Thougts?

Reply:
I think a resource helper is great improvement here.